### PR TITLE
context menu: do not recursively check pin and availability states

### PR DESCRIPTION
**Pull request by mgallien**

### CODE REVIEW

File: src/common/vfs.h [hunk 1]
```diff
@@ -258,7 +265,7 @@ class OCSYNC_EXPORT Vfs : public QObject
      *
      * folderPath is relative to the sync folder. Can be "" for root folder.
      */
-    Q_REQUIRED_RESULT virtual AvailabilityResult availability(const QString &folderPath) = 0;
+    Q_REQUIRED_RESULT virtual AvailabilityResult availability(const QString &folderPath, AvailabilityRecursivity recursiveCheck) = 0;
```
Comment (claucambra) [inline src/common/vfs.h:268]: nitpick: constify new arg?

File: src/common/vfs.h [hunk 2]
```diff
@@ -328,7 +335,7 @@ class OCSYNC_EXPORT VfsOff : public Vfs
 
     bool setPinState(const QString &, PinState) override { return true; }
     Optional<PinState> pinState(const QString &) override { return PinState::AlwaysLocal; }
-    AvailabilityResult availability(const QString &) override { return VfsItemAvailability::AlwaysLocal; }
+    AvailabilityResult availability(const QString &, AvailabilityRecursivity) override { return VfsItemAvailability::AlwaysLocal; }
```
Comment (claucambra) [inline src/common/vfs.h:338]: nitpick: constify new arg?

**Review by claucambra — APPROVED** [review]
Tiny nitpick comments but otherwise LGTM

File: src/common/vfs.h (hunk 1 above)
Comment (mgallien) [reply src/common/vfs.h]: done

**Pull request merged**